execute/executor.go

```diff
@@ -314,11 +320,15 @@ func (es *executionState) do(ctx context.Context) {
 			es.abort(err)
 		}
 	}()
+
+	go func() {
+		defer close(es.metaCh)
+		wg.Wait()
+	}()
```

Comment:
this change makes it so that we don't close the meta channel until the wait group has finished, which means that the dispatcher has stopped?

Reply:
That's my current understanding. And the query waits for the meta channel to close before declaring itself done.